code/modules/mob/login.dm

```diff
@@ -115,6 +115,16 @@
 				var/spell/S = screen.spell
 				mind.learned_spells |= S
 
+	var/show_status_markers = get_preference_value(/datum/client_preference/show_status_markers) == GLOB.PREF_SHOW
+	for(var/obj/status_marker_holder/marker as anything in global.status_marker_holders)
+		client.images -= marker.mob_image
+		client.images -= marker.mob_image_personal
```

Comment:
Why do we do this if the preference is don't show? They should have been removed on pref change or otherwise never added.

Reply:
I figured it was harmless and the extra check wasn't really necessary. I can add it back in if you like.

Reply:
Well, it's in Login so on the one hand doesn't matter; on the other hand having the whole loop not be run only if the pref is to show makes it strange. Don't care either way.

Reply:
Right, misunderstood, the purpose of this loop is just to refresh client images. I pruned out the `-=` and tidied the block a bit.